config.nix

```
@@ -349,7 +349,14 @@ let
     toplevelCargotoml = readTOML (root + "/Cargo.toml");
 
     # The cargo lock
-    cargolock = readTOML (root + "/Cargo.lock");
+    cargolock = 
+      let
+        cargolock-file = root + "/Cargo.lock";
+      in
+      if builtins.pathExists cargolock-file then
+        readTOML (cargolock-file)
+      else 
+        throw "Naersk requires Cargo.lock to be available in root. Check that it is not in .gitignore and stage it when using git to filer (which flakes does)";
```

nit:

```suggestion
        throw "Naersk requires Cargo.lock to be available in root. Check that it is not in .gitignore and stage it when using git to filter (which flakes does)";
```

I'm not sure what using "stage it when using to filter" means.

---

stage it with `git add Cargo.lock` since flake filters the source using gitignore